### PR TITLE
[FIP-41] LUSD Auction

### PR DESCRIPTION
**Pull request by eswak**
Auction 100M FEI for LUSD using a Balancer LBP, over 2 weeks.

Note: Fei Labs will spend 1,100,000 FEI to get LUSD, and send those to the swapper.
The DAO proposal includes a refund of this FEI.

Snapshot: https://snapshot.org/#/fei.eth/proposal/0x0e5f05a0c51938b904d9932849251ae920403b75301f90567da9d1ed857965c3

⚠️ I'm not 100% sure of the fLUSD-91 token address (`0x5052BfbB7972E702179f3Eeed43B9213819b681a`), to be confirmed with Liquity/Rari team.

### CODE REVIEW

File: proposals/dao/fip_41.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+chai.use(CBN(ethers.BigNumber));
+
+// Constants
+// LBP swapper
+const MIN_LBP_SIZE = ethers.constants.WeiPerEther.mul(50_000); // 50k FEI
+
+/*
+
+TRIBE Buybacks
```
Comment (ditchfieldcaleb) [inline proposals/dao/fip_41.ts:23]: Comment is wrong

File: proposals/dao/fip_41.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+6. Start auction
+*/
+
+export const deploy: DeployUpgradeFunc = async (deployAddress, addresses, logging = false) => {
+  const { core, fei, lusd, oneConstantOracle, balancerLBPoolFactory } = addresses;
+
+  if (!core) {
+    console.log(`core: ${core}`);
+
+    throw new Error('An environment variable contract address is not set');
```
Comment (ditchfieldcaleb) [inline proposals/dao/fip_41.ts:47]: nit: this isn't an environment variable.

File: proposals/description/fip_41.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        '1089000000000000000000000', // minDy
+        '1640995200' // deadline: 2022-01-01
+      ],
+      description: 'Swap 1.1M FEI for LUSD'
+    },
+    {
+      target: 'lusd',
+      values: '0',
+      method: 'transfer(address,uint256)',
+      arguments: ['{feiLusdLBPSwapper}', '1089000000000000000000000'],
```
Comment (ditchfieldcaleb) [inline proposals/description/fip_41.ts:37]: we'll need to make sure we submit the dao-execute transaction via flashbots protect, or someone could grief this swap

**Review by ditchfieldcaleb — DISMISSED** [review]
Overall it looks fine, but for a PR with potential impact of this magnitude, I'd like to specifically see some integration tests (even if they're throwaway-after-this-is-executed!) that test swaps after the PR executes.

File: proposals/dao/fip_41.ts (hunk 2 above)
Comment (Joeysantoro) [reply proposals/dao/fip_41.ts:47]: prob leave as is

File: proposals/dao/fip_41.ts (hunk 1 above)
Comment (Joeysantoro) [reply proposals/dao/fip_41.ts:23]: fixed

File: test/integration/tests/buybacks.ts [hunk 4]
```diff
@@ -78,6 +85,66 @@ describe('e2e-buybacks', function () {
...
+        },
+        {
+          sender: LUSD_HOLDING_ADDRESS,
+          fromInternalBalance: false,
+          recipient: LUSD_HOLDING_ADDRESS,
+          toInternalBalance: false
+        },
+        ethers.constants.WeiPerEther.mul(200_000),
+        ethers.constants.WeiPerEther // huge deadline
+      );
```
Comment (ditchfieldcaleb) [inline test/integration/tests/buybacks.ts:117]: would be good to check balances in between each swap. doesn't have to be exact, just asset that the balance of the swapped-in asset increases in the bool and the balance of the swapped-out asset decreases.

Comment (Joeysantoro) [reply test/integration/tests/buybacks.ts:117]: added test

**Pull request merged**